File: Kotlin/052.kt
fun main() {
	val N = readLine()!!.toInt()

	val m = 1000000007L

	val A = (0 until N).map { readLine()!!.split(' ').map(String::toLong) }

	val answer = A.fold(1L) { acc, list ->
		(acc * list.sum()) % m
	}

	println(answer)
}

File: Swift/052.swift
let N = Int(readLine()!)!

let m = 1000000007

let A = (0 ..< N).map { i in readLine()!.split(separator: " ").map { Int($0)! } }

let answer = A.reduce(1) { acc, list in
	(acc * list.reduce(0, +)) % m
}

print(answer)
